pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -39,8 +39,8 @@ def _basic_input_validation(preds: torch.Tensor, target: torch.Tensor, threshold
     if preds_float and (preds.min() < 0 or preds.max() > 1):
         raise ValueError("The `preds` should be probabilities, but values were detected outside of [0,1] range.")
 
-    if threshold > 1 or threshold < 0:
-        raise ValueError("The `threshold` should be a probability in [0,1].")
+    if not 0 < threshold < 1:
+        raise ValueError(f"The `threshold` should be a float in the (0,1) interval, got {threshold}")
```

Comment:
why is this checked again?

Reply:
So the threshold is checked once on input formatting, and for class metrics it is also checked at init (this was suggested in accuracy PR)